src/saveload/saveload.cpp

```diff
@@ -909,23 +909,74 @@ static inline size_t SlCalcStdStringLen(const void *ptr)
 
 
 /**
- * Scan the string for old values of SCC_ENCODED and fix it to it's new, value.
+ * Scan the string for old values of SCC_ENCODED and fix it to it's new value.
  * Note that at the moment this runs, the string has not been validated yet
  * because the validation looks for SCC_ENCODED. If there is something invalid,
  * just bail out and do not continue trying to replace the tokens.
  * @param str the string to fix.
  */
-static void FixSCCEncoded(std::string &str)
+void FixSCCEncoded(std::string &str, bool fix_code)
 {
-	for (size_t i = 0; i < str.size(); /* nothing. */) {
-		size_t len = Utf8EncodedCharLen(str[i]);
-		if (len == 0 || i + len > str.size()) break;
+	if (str.empty()) return;
+
+	/* We need to convert from old escape-style encoding to record separator encoding.
+	 * Initial `<SCC_ENCODED><STRINGID>` stays the same.
+	 *
+	 * `:<SCC_ENCODED><STRINGID>` becomes `<RS><SCC_ENCODED><STRINGID>`
+	 * `:<HEX>`                   becomes `<RS><SCC_ENCODED_NUMERIC><HEX>`
+	 * `:"<STRING>"`              becomes `<RS><SCC_ENCODED_STRING><STRING>`
+	 */
+	std::string result;
+	auto output = std::back_inserter(result);
+
+	bool isencoded = false; // Set if we determine by the presence of SCC_ENCODED that the string is an encoded string.
+	bool instring = false; // Set if we in a string, between double-quotes.
+	bool needtype = true; // Set if a parameter type needs to be emitted.
```

Comment:
```suggestion
	bool is_encoded = false; // Set if we determine by the presence of SCC_ENCODED that the string is an encoded string.
	bool in_string = false; // Set if we in a string, between double-quotes.
	bool need_type = true; // Set if a parameter type needs to be emitted.
```
Isn't it coding style to separate words by underscore?